src/app/shared/utilities/parsers.ts

```diff
@@ -0,0 +1,53 @@
+import Parser from 'morph-expressions'
+
+export function parseVersion(data) {
+  // NOTE: Based on current Google Play Store page pattern
+  // TODO: Add support for iOS
+  const playstorePattern = /<span class="htlgb">([0-9]*[.][0-9]*[.][0-9]*)<\/span>/g
+  const versionPattern = /([0-9]*[.][0-9]*[.][0-9]*)/
+  return data
+    .match(playstorePattern)
+    .toString()
+    .match(versionPattern)[0]
+}
+
+// NOTE: Parses and evaluates the branching logic
+// Example: '[esm_social_interact(1)] = \"1\" or' or '[esm_social] = '1'' to 'esm_social_interact[1] == "1" ||'
+export function parseAndEvalLogic(logic: string, answers): string {
```

Comment:
will this work fine with empty logic?

Reply:
Yup it checks for `logic.length` before going here